src/components/Navigation/components/Item/Item.tsx

```diff
@@ -40,7 +40,7 @@ interface SecondaryAction {
 export interface Props extends ItemURLDetails {
   icon?: IconProps['source'];
   iconBody?: string;
-  badge?: string | null;
+  badge?: string | React.ReactNode | null;
```

Comment:
`string | null` are included in `React.ReactNode` so I think you are OK to just use that type